server/src/proxy/grpc/write.rs

```diff
@@ -167,26 +169,44 @@ pub async fn write_request_to_insert_plan<Q: QueryExecutor + 'static>(
         deadline,
         auto_create_table,
     } = write_context;
-
+    let schema_config = schema_config.cloned().unwrap_or_default();
     for write_table_req in table_requests {
         let table_name = &write_table_req.table;
         let mut table = try_get_table(&catalog, &schema, instance.clone(), table_name)?;
 
-        if table.is_none() && auto_create_table {
-            // TODO: remove this clone?
-            let schema_config = schema_config.cloned().unwrap_or_default();
-            create_table(
-                request_id,
-                &catalog,
-                &schema,
-                instance.clone(),
-                &write_table_req,
-                &schema_config,
-                deadline,
-            )
-            .await?;
-            // try to get table again
-            table = try_get_table(&catalog, &schema, instance.clone(), table_name)?;
+        match table.clone() {
+            None => {
+                if auto_create_table {
+                    create_table(
+                        request_id,
+                        &catalog,
+                        &schema,
+                        instance.clone(),
+                        &write_table_req,
+                        &schema_config,
+                        deadline,
+                    )
+                    .await?;
+                    // try to get table again
+                    table = try_get_table(&catalog, &schema, instance.clone(), table_name)?;
+                }
+            }
+            Some(t) => {
```

Comment:
We should only alter schema when write failed with specific error code.

Current changes will be invoked at every write(hot path), which is not ideal way to do this.

Reply:
* If judged based on the error message returned, it may cause data that has already been successfully written to be written again and affect the accuracy of the data.
* Currently, the decision to add columns is made at the request level, not at the row level, so the cost is relatively small.